### ISSUE TITLE
Editor can't be launched on Windows when using full path to editor executable

### ISSUE THREAD

**Issue by cs-ghi** [jrnl-org/jrnl#1096]
### Environment

  - Jrnl `--diagnostic` output: 
jrnl: v2.5
Python: 3.8.6 (tags/v3.8.6:db45529, Sep 23 2020, 15:52:53) [MSC v.1927 64 bit (AMD64)]
OS: Windows 10

  - Install method: <!-- How did you install jrnl? 
pipx

### What are you trying to do?
I am trying to change my editor to vim.

### What have you tried?
I changed the value of the "editor" key on the yaml file to "vim", "gvim", "Vim", "C:\\tools\\vim\\vim82\\vim.exe" - none of them worked.
Then I tried to follow the instructions on the "recipes" page on the documentation and tried to set it to notepad++, just to see if it would work. It didn't work, either. (The path to my notepad++ executable is right)

```
[WinError 2] The system cannot find the file specified.

Please check the 'editor' key in your config file for errors:
'C:\\Program Files (x86)\\Notepad++\\notepad++.exe -multiInst -nosession'
```

Then I tried with a bunch of other text editors - the only one which worked was notepad.

### Other Information
If applicable, please run jrnl with --debug and paste the output.
(The editor here is still set to notepad++, hence the error)

```
DEBUG    root         Parsed args: Namespace(contains=None, debug=True, delete=False, edit=False, end_date=None, excluded=[], export=False, filename=None, limit=None, on_date=None, postconfig_cmd=None, preconfig_cmd=None, short=False, starred=False, start_date=None, strict=False, tags=False, text=[])
DEBUG    root         Reading configuration from file C:\Users\User\.config\jrnl\jrnl.yaml
DEBUG    root         Using configuration "{'colors': {'body': 'none', 'date': 'none', 'tags': 'none', 'title': 'none'}, 'default_hour': 9, 'default_minute': 0, 'editor': 'C:\\Program Files (x86)\\Notepad++\\notepad++.exe -multiInst -nosession', 'encrypt': False, 'highlight': True, 'indent_character': '|', 'journals': {'default': 'C:\\Users\\User\\Desktop\\Journal\\journal.txt', 'b': 'C:\\Users\\User\\Desktop\\Journal\\buddhism.txt'}, 'linewrap': 79, 'tagsymbols': '@', 'template': False, 'timeformat': '%Y-%m-%d %H:%M', 'version': 'v2.5'}"
DEBUG    root         Using journal name: default
DEBUG    root         opened PlainJournal with 12 entries
DEBUG    root         Write mode: starting
DEBUG    root         Write mode: opening editor
DEBUG    root         Write mode: loading template for entry
DEBUG    root         Write mode: no template configured
[WinError 2] The system cannot find the file specified

Please check the 'editor' key in your config file for errors:
    'C:\\Program Files (x86)\\Notepad++\\notepad++.exe -multiInst -nosession'
```

**Comment by cs-ghi** [comment]
Update: it worked when I copied and pasted the Vim.exe file in my %windir%\system32\ folder and then changed the value of the 'editor' key to "Vim"

**Comment by micahellison** [comment]
Thanks for the post and the update. I think you have found a Windows compatibility bug in jrnl.

I'm trying to reproduce this, and the editor key works for me when the exe is in the path (just like it does for you), but not when I spell out the whole path.

The problem appears to be that it's transforming backslashes into two backslashes.

My editor key:
```
C:\Program Files\Sublime Text 3\subl.exe -w
```
My error message:
```
[WinError 2] The system cannot find the file specified

Please check the 'editor' key in your config file for errors:
    'C:\\Program Files\\Sublime Text 3\\subl.exe -w'
```
I'll see about getting a fix for this in our next release.

**Comment by claviger-pc** [comment]
for me, I switched to single quotes with the doubled backslashes and this resolves the problem

**Comment by KarimPwnz** [comment]
Turns out the problem is in `shlex.split`:
```python3
>>> shlex.split("C:\Program Files\Sublime Text 3\subl.exe -w")
['C:Program', 'FilesSublime', 'Text', '3subl.exe', '-w']
```

**Comment by KarimPwnz** [comment]
It does not handle the spaces correctly!

**Comment by KarimPwnz** [comment]
Seems like numpy had to deal with the same issue (https://github.com/numpy/numpy/pull/13023). We could definitely use their parser, but I am not sure if using another package's utils is ideal.

**Comment by micahellison** [comment]
This was fixed by #1153.
